src/notebooks/controllers/vscodeNotebookController.ts

```diff
@@ -295,7 +297,11 @@ export class VSCodeNotebookController implements Disposable, IVSCodeNotebookCont
         }
         await initializeInteractiveOrNotebookTelemetryBasedOnUserAction(notebook.uri, this.connection);
         // Notebook is trusted. Continue to execute cells
-        await Promise.all(cells.map((cell) => this.executeCell(notebook, cell)));
+        await Promise.all(cells.map(async (cell) => {
+            await this.lock.acquireAsync();
```

Comment:
Thanks for the PR, I'll take a look at this.
Though this fixes the issue, we don't end up running `executeCell` for the other cells, when we should.
E.g. we need to wait for other cells to change state, I don't think this PR will ensure other cells go into a pending state.
Once again thanks & I'll take a look into this.
Will also file an issue so we can track the bug there.

Reply:
Yup, I agree with that. I just wanted to flag the bug and demo a fix. This can be done within executeCell as well, just that I am not familiar enough with the source code to decide which is best.

Reply:
FYI - Filed an issue here https://github.com/microsoft/vscode-jupyter/issues/11526
I'll try to submit changes to this (your) PR, thanks, I can confirm this is a bug.